File: pages/EmployeesPage.js
class EmployeesPage {
    constructor(page) {
        this.page = page;
        this.employeesSelectors = [
            'a:has-text("Employees")',
            '[href*="employees"]',
            'a:has-text("Employee")',
            'a[href*="/hr/"]',
            '.sidebar a:has-text("Employee")',
            'nav a:has-text("Employee")'
        ];
        this.addNewEmployeeButton = 'button:has-text("Add New Employee"), a:has-text("Add New Employee")';
    }

    async navigateToEmployees() {
        // Try each selector until we find one that works
        let found = false;
        for (const selector of this.employeesSelectors) {
            try {
                await this.page.waitForSelector(selector, { state: 'visible', timeout: 20000 });
                await this.page.click(selector);
                found = true;
                break;
            } catch (e) {
                continue;
            }
        }
        
        if (!found) {
            throw new Error('Could not find Employees link');
        }
        
        await this.page.waitForLoadState('networkidle', { timeout: 60000 });
    }

    async clickAddNewEmployee() {
        await this.page.waitForSelector(this.addNewEmployeeButton, { state: 'visible', timeout: 60000 });
        await this.page.click(this.addNewEmployeeButton);
        await this.page.waitForLoadState('networkidle', { timeout: 60000 });
    }

    async validateAddEmployeeURL() {
        // Wait for both URL and form elements
        await Promise.all([
            this.page.waitForURL(/.*add.*/, { timeout: 60000 }),
            this.page.waitForSelector('form, .add-employee-form, .employee-form', { timeout: 60000 })
        ]);
    }
}

module.exports = EmployeesPage;

File: pages_py/employees_page.py
from playwright.sync_api import Page

class EmployeesPage:
    def __init__(self, page: Page):
        self.page = page
        self.employees_selectors = [
            'a:has-text("Employees")',
            '[href*="employees"]',
            'a:has-text("Employee")',
            'a[href*="/hr/"]',
            '.sidebar a:has-text("Employee")',
            'nav a:has-text("Employee")'
        ]
        self.add_new_employee_button = 'button:has-text("Add New Employee"), a:has-text("Add New Employee")'

    def navigate_to_employees(self):
        found = False
        for selector in self.employees_selectors:
            try:
                self.page.wait_for_selector(selector, state='visible', timeout=20000)
                self.page.click(selector)
                found = True
                break
            except Exception:
                continue
        if not found:
            raise Exception('Could not find Employees link')
        self.page.wait_for_load_state('networkidle', timeout=60000)

    def click_add_new_employee(self):
        self.page.wait_for_selector(self.add_new_employee_button, state='visible', timeout=60000)
        self.page.click(self.add_new_employee_button)
        self.page.wait_for_load_state('networkidle', timeout=60000)

    def validate_add_employee_url(self):
        print("Debug: Current URL during validation:", self.page.url)
        # Validate by checking for specific elements on the Add Employee page
        self.page.wait_for_selector('form, .add-employee-form, .employee-form', timeout=90000)